File: integration-tests/android/src/main/java/com/example/ExampleActivity.java
package com.example;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.ViewGroup.LayoutParams;
import android.widget.TextView;
import dagger.android.AndroidInjection;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasAndroidInjector;

import javax.inject.Inject;

import static android.util.TypedValue.COMPLEX_UNIT_DIP;
import static android.view.Gravity.CENTER;
import static android.view.ViewGroup.LayoutParams.MATCH_PARENT;

public final class ExampleActivity extends Activity implements HasAndroidInjector {
    @Inject
    DispatchingAndroidInjector<Object> androidInjector;

    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);

        TextView textView = new TextView(this);
        textView.setLayoutParams(new LayoutParams(MATCH_PARENT, MATCH_PARENT));
        textView.setGravity(CENTER);
        textView.setTextSize(COMPLEX_UNIT_DIP, 40);
        textView.setText("hello");
        setContentView(textView);
        getFragmentManager().beginTransaction().add(new ExampleFragment(), "some tag").commit();
    }

    @Override
    public AndroidInjector<Object> androidInjector() {
        return androidInjector;
    }
}

File: integration-tests/android/src/main/java/com/example/ExampleActivity.kt
package com.example

import android.os.Bundle
import android.util.Log
import android.view.ViewGroup.LayoutParams
import android.widget.TextView

import javax.inject.Inject

import android.util.TypedValue.COMPLEX_UNIT_DIP
import android.view.Gravity.CENTER
import android.view.ViewGroup.LayoutParams.MATCH_PARENT
import javax.inject.Provider

class ExampleActivity : BaseActivity<ExamplePresenter, ExampleActivity.FooReal>() {
      @Inject lateinit var string: String

    @Inject lateinit var providerString: Provider<String>

    @Inject
    lateinit var exampleClass: ExampleClass

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val textView = TextView(this)
        textView.layoutParams = LayoutParams(MATCH_PARENT, MATCH_PARENT)
        textView.gravity = CENTER
        textView.setTextSize(COMPLEX_UNIT_DIP, 40f)
        textView.text = providerString.get();
        setContentView(textView)

        Log.e("testaaaa", toString())
        exampleClass!!.bar()
        presenterDelegate.getPresenter().foo()
        // startService(new Intent(this, ExampleService.class));
        addExampleFragment()
    }

    private fun addExampleFragment() {
        supportFragmentManager.beginTransaction().add(ExampleFragment(), ExampleActivity::class.java.simpleName).commit()
    }

    class FooReal : BaseActivity.Foo
}
